src/classes/CW_CampaignRewardCompControllerTest.cls

```diff
@@ -0,0 +1,260 @@
+		system.assertEquals(0     ,compCtrl.getSelectedCount());
+			
+		compCtrl.last();
+		system.assertEquals(1     ,compCtrl.getRewardList().Size());
+			
+		compCtrl.doNext();
+		system.assertEquals(2     ,compCtrl.getPageNumber());
+		system.assertEquals(true  ,compCtrl.getHasPrevious());
+		system.assertEquals(false ,compCtrl.getHasNext());
+			
+		compCtrl.doPrevious();
+		system.assertEquals(false  ,compCtrl.getHasPrevious());
+		system.assertEquals(true   ,compCtrl.getHasNext());
+		
+		compCtrl.selectedRewardIds.Add(listRewards[3].Id);
+		compCtrl.selectedRewardIds.Add(listRewards[4].Id);
+		system.assertEquals(false  ,compCtrl.getDeleteDisabled());
+		compCtrl.getDeleteMessage();
+		compCtrl.deleteRewards();
+
+		system.assertEquals(9     ,compCtrl.getRewardList().Size());
+		system.assertEquals(false  ,compCtrl.editMode);
+
+		compCtrl.savedByParent =true;
+		system.assertEquals(false  ,compCtrl.editMode);
+		compCtrl.Cancel();
+		system.assertEquals(false  ,compCtrl.editMode);
+	}
+	
+	static testMethod void createCampaignReward() 
+	{
+		initCustomSettings();
+		
+		CW_Campaign__c camp = TestDataUtil.createCampaign();
+		insert camp;
+		
+		CW_Offer_Type__c offerType = TestDataUtil.createOfferType('Operational Tool');
+		insert offerType;
+		
+		CW_Reward_Type__c rewardType = TestDataUtil.createRewardType('FreeBet Amount');
+		insert rewardType;
+		
+		CW_Reward_Type__c cashRewardType = TestDataUtil.createRewardType('Cash');
+		insert cashRewardType;
+		
+		CW_Reward_Catalogue__c rewardCat = TestDataUtil.createRewardCatalogue(offerType.Id,rewardType.Id);
+		rewardCat.Library__c = true;
+		insert rewardCat;
+		
+		rewardCat = TestDataUtil.createRewardCatalogue(offerType.Id,cashRewardType.Id);
+		insert rewardCat;
+		
+		CW_Reward_Library__c rewardLib= TestDataUtil.createRewardLibrary(rewardType.Id);
+		insert rewardLib;
+		
+		CW_Campaign_Offer__c campOffer = TestDataUtil.createCampaignOffer(camp.Id, offerType.Id);
+		insert campOffer; 
+		
+		CW_CampaignRewardCompController compCtrl= new CW_CampaignRewardCompController();
+    	compCtrl.campOfferId = campOffer.Id;
+    	compCtrl.campOffer = campOffer;
+    	compCtrl.offerTypeId = offerType.Id;
+    	
+    	compCtrl.addReward();
+    	system.assertEquals(true,compCtrl.NewRecMode);
+    	compCtrl.getRewardTypeOptions();
+
+    	compCtrl.RewardTypeId = rewardType.Id;
+    	compCtrl.getFields();
+    	compCtrl.campOfferOrRewardHasLibrary();
+    	system.assertEquals('OperationalToolFreeBetAmountFieldSet',compCtrl.fieldsetName,'Wrong fieldset');
+    	system.assertEquals(true,compCtrl.hasLibrary,'Library lookup field is not displayed correctly');
+    	
+    	compCtrl.newRec.Reward_Type__c = cashRewardType.Id;
+    	compCtrl.hasLibrary = false;
+    	compCtrl.newRec.Currency__c = 'GBP';
+    	compCtrl.newRec.Amount__c = 200;
+    	compCtrl.SaveNewRecord();
+    	List<CW_Campaign_Reward__c> campRewards = [SELECT Id FROM CW_Campaign_Reward__c WHERE Campaign_Offer__c =: compCtrl.campOfferId];
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.